### PR TITLE
Cbagraphv1.0

### PR DESCRIPTION
**Pull request by vimrang**
Creating msgraph doc for v1.0 for X509 certificate based authentication

### CODE REVIEW

File: changelog/Microsoft.DirectoryServices.json [hunk 1]
```diff
@@ -1,5 +1,63 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the [x509CertificateUserBinding](https://docs.microsoft.com/en-us/graph/api/resources/x509CertificateUserBinding?view=graph-rest-1.0) resource type. Use this resource to map the fields in the X.509 certificate to attributes of the Azure AD user object in order to bind the certificate to the user account.",
+          "Target": "x509CertificateUserBinding"
+        },
+        {
+          "Id": "10c162ec-a1ea-4b26-a865-c4d54de08c4a",
+          "ApiChange": "Resource",
+          "ChangedApiName": "x509CertificateAuthenticationMethodConfiguration",
+          "ChangeType": "Addition",
+          "Description": "Added the [x509CertificateAuthenticationMethodConfiguration](https://docs.microsoft.com/en-us/graph/api/resources/x509CertificateAuthenticationMethodConfiguration?view=graph-rest-1.0) resource type. Represents the details of the Azure AD native Certificate-Based Authentication (CBA) in the tenant including whether the authentication method is enabled or disabled and the users and groups who are in scope for the auth method.",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:50]: nit: 
```suggestion
          "Description": "Added the [x509CertificateAuthenticationMethodConfiguration](https://docs.microsoft.com/en-us/graph/api/resources/x509CertificateAuthenticationMethodConfiguration?view=graph-rest-1.0) resource type and its associated methods. This resource represents the details of the Azure AD native Certificate-Based Authentication (CBA) in the tenant including whether the authentication method is enabled or disabled and the users and groups who are in scope for the auth method.",
```

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]
Thank you @vimrang for your work in this PR. I've left a few comments for minor fixes. Two things to note:

1. For every response, there must be a request. So you must have the http request blocks immediately after the HTML comment blocks. For example, this 👇🏾 is a complete "Request" section in Markdown.

            ### Request
            
            
            ```
            DELETE https://graph.microsoft.com/beta/policies/authenticationMethodsPolicy/authenticationMethodConfigurations/x509Certificate
            ```

2. Given this is the "policy/configuration", it's not cleear what the methods is. Also, is this API related to the [certificateBasedAuthConfiguration resource type and related methods](https://learn.microsoft.com/en-us/graph/api/resources/certificatebasedauthconfiguration?view=graph-rest-1.0). Let's follow up internally and have an end-to-end tutorial for using this new API and any related APIs.

File: api-reference/v1.0/resources/x509certificateuserbinding.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+Namespace: microsoft.graph
+
+Defines the fields in the X.509 certificate that map to attributes of the Azure AD user object in order to bind the certificate to the user account.
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|priority|Int32|The priority of the binding. Azure AD uses the binding with the highest priority. This value must be a non-negative integer and unique in the collection of objects in the **certificateUserBindings** property of an **x509CertificateAuthenticationMethodConfiguration** object. Required|
+|userProperty|String|Defines the Azure AD user property of the user object to use for the binding. The possible values are: **userPrincipalName**, `onPremisesUserPrincipalName`, `email`. Required.|
+|x509CertificateField|String|The field on the X.509 certificate to use for the binding. The possible values are: `PrincipalName`, `RFC822Name`.|
```
Comment (FaithOmbongi) [inline api-reference/v1.0/resources/x509certificateuserbinding.md:21]: Your API.md shows two additional possible values. Can you verify?

File: api-reference/v1.0/resources/x509certificateuserbinding.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+Namespace: microsoft.graph
+
+Defines the fields in the X.509 certificate that map to attributes of the Azure AD user object in order to bind the certificate to the user account.
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|priority|Int32|The priority of the binding. Azure AD uses the binding with the highest priority. This value must be a non-negative integer and unique in the collection of objects in the **certificateUserBindings** property of an **x509CertificateAuthenticationMethodConfiguration** object. Required|
+|userProperty|String|Defines the Azure AD user property of the user object to use for the binding. The possible values are: **userPrincipalName**, `onPremisesUserPrincipalName`, `email`. Required.|
```
[the remark this answers is not in the document]
Comment (vimrang) [reply api-reference/v1.0/resources/x509certificateuserbinding.md:20]: Corrected to use 4 properties for certificate field and 3 values for user object fields

File: api-reference/v1.0/toc.yml [hunk 4]
```diff
@@ -5000,6 +5000,16 @@ items:
...
+          - name: X509 certificate
+            displayName: certificate-based authentication, CBA
+            href: resources/x509Certificateauthenticationmethodconfiguration.md
+            items:
+            - name: Get
+              href: api/x509certificateauthenticationmethodconfiguration-get.md
+            - name: Update
+              href: api/x509certificateauthenticationmethodconfiguration-update.md
+            - name: Delete
+              href: api/x509certificateauthenticationmethodconfiguration-delete.md
```
[the remark this answers is not in the document]
Comment (vimrang) [reply api-reference/v1.0/toc.yml:5012]: moved

File: api-reference/v1.0/api/x509certificateauthenticationmethodconfiguration-delete.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+### Request
+
+# [HTTP](#tab/http)
+<!-- {
+  "blockType": "request",
+  "name": "delete_x509certificateauthenticationmethodconfiguration"
+}
+-->
+
```
[the remark this answers is not in the document]
Comment (vimrang) [reply api-reference/v1.0/api/x509certificateauthenticationmethodconfiguration-delete.md:61]: corrected

File: api-reference/v1.0/api/x509certificateauthenticationmethodconfiguration-get.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+## Examples
+
+### Request
+
+# [HTTP](#tab/http)
+<!-- {
+  "blockType": "request",
+  "name": "get_x509certificateauthenticationmethodconfiguration"
+}
+-->
```
[the remark this answers is not in the document]
Comment (vimrang) [reply api-reference/v1.0/api/x509certificateauthenticationmethodconfiguration-get.md:64]: corrected

File: api-reference/v1.0/api/x509certificateauthenticationmethodconfiguration-update.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+### Request
+
+The following is an example of an update request with the following settings:
+
++ Enables the x509 certificate authentication method in the tenant.
++ Configures only one user binding between the certificate **PrincipalName** and the Azure AD **onPremisesUserPrincipalName** properties.
++ Defines multi-factor authentication as requirement.
++ Configures the binding rules for the strong authentication method against the rule type.
+
+# [HTTP](#tab/http)
```
[the remark this answers is not in the document]
Comment (vimrang) [reply api-reference/v1.0/api/x509certificateauthenticationmethodconfiguration-update.md:73]: corrected

File: changelog/Microsoft.DirectoryServices.json (hunk 1 above)
Comment (vimrang) [reply changelog/Microsoft.DirectoryServices.json:50]: corrected

File: api-reference/v1.0/resources/x509certificateuserbinding.md (hunk 2 above)
Comment (vimrang) [reply api-reference/v1.0/resources/x509certificateuserbinding.md:21]: corrected

**Comment by Lauragra** [comment]
@vimrang , if you would like to publish this PR, please remove the **do not merge** label and add **ready to merge**.

**Pull request merged**